server/src/main/java/org/candlepin/resource/CrlResource.java

```diff
@@ -73,24 +79,23 @@ public CrlResource(CrlGenerator crlGenerator,
      */
     @GET
     @Produces({ MediaType.APPLICATION_JSON, MediaType.TEXT_PLAIN })
-    public String getCurrentCrl(@Context Principal principal)
+    public Response getCurrentCrl(@Context Principal principal)
         throws CRLException, IOException {
 
         String filePath = getCrlFilePath();
         File crlFile = new File(filePath);
 
-        byte[] encoded = null;
-
         try {
+            // TODO: We could probably just invoke the task here and not duplicate code
             X509CRL crl = crlFileUtil.readCRLFile(crlFile);
             crl = crlGenerator.syncCRLWithDB(crl);
-            encoded = crlFileUtil.writeCRLFile(crlFile, crl);
+            crlFileUtil.writeCRLFile(crlFile, crl);
         }
         catch (CertificateException e) {
             throw new IseException(e.getMessage(), e);
         }
 
-        return new String(encoded);
+        return Response.ok().entity(new FileInputStream(crlFile)).build();
```

Comment:
Do we need to be stricter about the content type set by the Produces annotation here?  As I read it, the content type will always be text/plain so having application/json in there might not be appropriate?